core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/avro/AvroInlet.scala

```diff
@@ -40,6 +41,18 @@ object AvroInlet {
   def create[T <: SpecificRecordBase](name: String, clazz: Class[T]): AvroInlet[T] =
     AvroInlet[T](name)(ClassTag.apply(clazz))
 
+  // Java API
   def create[T <: SpecificRecordBase](name: String, clazz: Class[T], hasUniqueGroupId: Boolean): AvroInlet[T] =
     AvroInlet[T](name, hasUniqueGroupId)(ClassTag.apply(clazz))
+
+  // Java API
+  def create[T <: SpecificRecordBase](
+      name: String,
+      clazz: Class[T],
+      hasUniqueGroupId: Boolean,
+      errorHandler: (Array[Byte], Throwable) => Optional[T]): AvroInlet[T] =
+    AvroInlet[T](name, hasUniqueGroupId, (a, t) => {
+      val opt = errorHandler(a, t)
+      if (opt.isPresent) Some(opt.get()) else None
```

Comment:
Isn't this same as returning the `opt` itself ?

Reply:
LOL I must have refactored this somehow.. good catch!

Reply:
It's calling the Scala constructor which needs `Option`, not `Optional`